src/textual/app.py

```diff
@@ -892,23 +893,52 @@ async def _on_css_change(self) -> None:
     def render(self) -> RenderableType:
         return Blank(self.styles.background)
 
+    ExpectType = TypeVar("ExpectType", bound=Widget)
+
+    @overload
     def get_child_by_id(self, id: str) -> Widget:
+        ...
+
+    @overload
+    def get_child_by_id(self, id: str, expect_type: type[ExpectType]) -> ExpectType:
+        ...
+
+    def get_child_by_id(
+        self, id: str, expect_type: type[ExpectType] | None = None
+    ) -> ExpectType | Widget:
         """Shorthand for self.screen.get_child(id: str)
         Returns the first child (immediate descendent) of this DOMNode
         with the given ID.
 
         Args:
             id (str): The ID of the node to search for.
+            expect_type (type | None, optional): Require the object be of the supplied type, or None for any type.
+                Defaults to None.
 
         Returns:
-            DOMNode: The first child of this node with the specified ID.
+            ExpectType | Widget: The first child of this node with the specified ID.
 
         Raises:
             NoMatches: if no children could be found for this ID
+            WrongType: if the wrong type was found.
         """
-        return self.screen.get_child_by_id(id)
```

Comment:
Do we need this condition? The second parameter can be None can’t it?

Reply:
I was trying to copy what is done here for `query_one`:
https://github.com/Textualize/textual/blob/f7867178a310a0e577c760a0ef3be0b0cab54316/src/textual/dom.py#L758

And noticed that when I tried:
```python
return self.screen.get_child_by_id(id, expect_type=expect_type)
```

Pylance was yelling at me with:
> Return type is unknown

So I decided it was best to follow the `if/else` pattern. Would really love to learn more about the decision either way though, a lot of this is still new to me.

Reply:
I’m not sure why either. But not fussed either way.